hazelcast/src/main/java/com/hazelcast/util/ExceptionUtil.java

```diff
@@ -48,23 +48,39 @@ public static String toString(Throwable cause) {
         return sw.toString();
     }
 
+    public static RuntimeException peel(final Throwable t) {
+        return (RuntimeException) peel(t, null);
+    }
+
+    public static <T extends Throwable> Throwable peel(final Throwable t, Class<T> allowedType) {
+        if (t instanceof RuntimeException) {
+            return t;
+        }
+
+        if (t instanceof ExecutionException) {
+            final Throwable cause = t.getCause();
+            if (cause != null) {
+                return peel(cause, allowedType);
+            } else {
+                return new HazelcastException(t);
+            }
+        }
+
+        if (allowedType != null && allowedType.isAssignableFrom(t.getClass())) {
+            return t;
+        }
+
+        return new HazelcastException(t);
```

Comment:
I think the whole design needs a careful review. It is OK to have a method which unchecks an exception; what is not OK is to use this method indiscriminately. We'll probably need one more utility method which will `throws InterruptedException` and uncheck everything else.